plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java

```diff
@@ -196,7 +200,8 @@ private IcebergFileWriter createParquetWriter(
                     getCompressionCodec(session).getParquetCompressionCodec(),
                     nodeVersion.toString(),
                     outputPath,
-                    fileSystem);
+                    fileSystem,
```

Comment:
make `IcebergParquetFileWriter` accept `TrinoOutputFile` rather than `OutputStream`. Similar for `ParquetFileWriter`.

Please make it a pre-commit